axelor-stock/src/main/java/com/axelor/apps/stock/web/PickedProductController.java

```diff
@@ -21,4 +25,16 @@ public void setFromStockLocationDomain(ActionRequest request, ActionResponse res
               .getStockLocationDomain(pickedProduct, massStockMove));
     }
   }
+
+  public void realizePicking(ActionRequest request, ActionResponse response)
+      throws AxelorException {
+    var pickedProduct =
+        Optional.of(request.getContext().asType(PickedProduct.class))
+            .map(pp -> Beans.get(PickedProductRepository.class).find(pp.getId()))
+            .orElse(null);
```

Comment:
I might be wrong, but if `pickedProduct` is null, we will get a runtime error from `realize` ? Then what is the point of the combo of optional + orElse(null) ?